src/app/options.cpp

```diff
@@ -235,6 +240,69 @@ namespace
         return o == s;
     }
 
+    // Option that is explicitly set to true or false, and whose value is undefined when unspecified.
+    // May not have a shortcut.
+    class TriStateBoolOption: protected StringOption
+    {
+    public:
+        constexpr TriStateBoolOption(const char *name)
+            : StringOption {name}
+        {
+        }
+
+        using StringOption::operator==;
+        
+        QString usage() const
+        {
+            return StringOption::usage(QLatin1String("true|false"));
+        }
+
+        TriStateBool value(const QString &arg) const
+        {
+            QString val = StringOption::value(arg);
+            
+            if (val.toUpper() == QLatin1String("TRUE") || val == QLatin1String("1")) {
+                return TriStateBool::True;
+            }
+            else if (val.toUpper() == QLatin1String("FALSE") || val == QLatin1String("0")) {
+                return TriStateBool::False;
+            }
+            else {
+                //: Translation must use English "true" and "false"
```

Comment:
The second argument of `QObject::tr()` is meant for disambiguation, like when you have two strings that are the same in English but might need to be translated differently in another language. You're supposed to use the `//:` comment notation for notes to the translator. See [this section of the Qt docs on translations](http://doc.qt.io/qt-5/i18n-source-translation.html#translator-comments)